client/web/src/insights/pages/dashboards/dashboard-page/components/dashboard-menu/DashboardMenu.tsx

```diff
@@ -73,3 +85,90 @@ export const DashboardMenu: React.FunctionComponent<DashboardMenuProps> = props
         </Menu>
     )
 }
+
+enum DashboardReasonDenied {
+    BuiltInCantBeEdited,
+    PermissionDenied,
+    UnknownDashboard,
+}
+
+type DashboardPermissions =
+    | {
+          isConfigurable: false
+          reason: DashboardReasonDenied
+      }
+    | {
+          isConfigurable: true
+      }
+
+const DEFAULT_DASHBOARD_PERMISSIONS: DashboardPermissions = {
+    isConfigurable: false,
+    reason: DashboardReasonDenied.UnknownDashboard,
+}
+
+function useDashboardPermissions(
+    dashboard: InsightDashboard | undefined,
+    settingsCascade: SettingsCascadeOrError<Settings>
+): DashboardPermissions {
+    const supportedSubject = useInsightSubjects({ settingsCascade })
+
+    if (isVirtualDashboard(dashboard)) {
+        return {
+            isConfigurable: false,
+            reason: DashboardReasonDenied.BuiltInCantBeEdited,
+        }
+    }
+
+    const dashboardOwner = supportedSubject.find(subject => subject.id === dashboard?.owner?.id)
+
+    // No dashboard can't be modified
+    if (!dashboard || !dashboardOwner) {
+        return DEFAULT_DASHBOARD_PERMISSIONS
+    }
+
+    if (isRealDashboard(dashboard)) {
+        // Settings based insights dashboards (custom dashboards created by users)
+        if (isSettingsBasedInsightsDashboard(dashboard)) {
+            // Global scope permission handling
+            if (isGlobalSubject(dashboardOwner)) {
+                const canBeEdited = dashboardOwner.viewerCanAdminister && dashboardOwner.allowSiteSettingsEdits
+
+                if (canBeEdited) {
+                    return { isConfigurable: true }
+                }
+
+                return {
+                    isConfigurable: false,
+                    reason: DashboardReasonDenied.PermissionDenied,
+                }
```

Comment:
```suggestion
                if (!canBeEdited) {
                    return {
                        isConfigurable: false,
                        reason: DashboardReasonDenied.PermissionDenied,
                    }
                }
```

(optional) possible that the current code is clearer since `canBeEdited` is more rare/an exceptional case, so it may be to explicitly return rather than defer to the next return